DDInstall/DDInstall.psm1

```diff
@@ -0,0 +1,192 @@
+    Comma-separated list of tags to assign in the configuration file. Example: TAGS="key_1:val_1,key_2:val_2"
+
+    .PARAMETER Hostname
+    Configures the hostname reported by the Agent to Datadog (overrides any hostname calculated at runtime).
+
+    .PARAMETER DDAgentUsername
+    Override the default ddagentuser username used during Agent installation (v6.11.0+).
+
+    .PARAMETER DDAgentPassword
+    Override the cryptographically secure password generated for the ddagentuser user during Agent installation (v6.11.0+). Must be provided for installs on domain servers.
+
+    .PARAMETER ApplicationDataDirectory
+    Override the directory to use for the configuration file directory tree (v6.11.0+). May only be provided on initial install; not valid for upgrades. Default: C:\ProgramData\Datadog
+
+    .PARAMETER ProjectLocation
+    Override the directory to use for the binary file directory tree (v6.11.0+). May only be provided on initial install; not valid for upgrades. Default: %ProgramFiles%\Datadog\Datadog Agent
+
+    .PARAMETER AgentInstallURL
+    Override the URL which the Agent installer is downloaded from.
+
+    .PARAMETER InstallLogPath
+    Override the Agent installation log location.
+
+    .INPUTS
+    None. You cannot pipe objects to Install-DDAgent.
+
+    .OUTPUTS
+    None.
+
+    .EXAMPLE
+    C:\PS> Install-DDAgent -ApiKey "<YOUR_DATADOG_API_KEY>"
+
+    .EXAMPLE
+    C:\PS> Install-DDAgent -ApiKey "<YOUR_DATADOG_API_KEY>" -WithAPMTracers "DotNet"
+
+    .LINK
+    Learn more about the Datadog Windows Agent User:
+    https://docs.datadoghq.com/agent/guide/windows-agent-ddagent-user/
+    #>
+}
+
+###############################
+# Unexported helper functions #
+###############################
+
+function formatAgentInstallerParameters($params)
+{
+    $formattedparams = "APIKEY=$($params.ApiKey)"
+
+    if ($params.ContainsKey('Site'))                        { $formattedparams += " SITE=$($params.Site)"}
+    if ($params.ContainsKey('Tags'))                        { $formattedparams += " TAGS=``$($params.Tags)``" }
+    if ($params.ContainsKey('Hostname'))                    { $formattedparams += " HOSTNAME=$($params.Hostname)" }
+    if ($params.ContainsKey('DDAgentUsername'))             { $formattedparams += " DDAGENTUSER_NAME=$($params.DDAgentUsername)" }
+    if ($params.ContainsKey('DDAgentPassword'))             { $formattedparams += " DDAGENTUSER_PASSWORD=$($params.DDAgentPassword)" }
+    if ($params.ContainsKey('ApplicationDataDirectory'))    { $formattedparams += " APPLICATIONDATADIRECTORY=$($params.ApplicationDataDirectory)" }
+    if ($params.ContainsKey('ProjectLocation'))             { $formattedparams += " PROJECTLOCATION=$($params.ProjectLocation)" }
+    if ($params.ContainsKey('InstallLogPath'))              { $formattedparams += " /log $($params.InstallLogPath)" }
+
+    return $formattedparams
+}
+
+function downloadAsset($url, $outFile)
+{
+    $ProgressPreference = 'SilentlyContinue'
```

Comment:
what is the scope of this option? Is it only for this function, or does it affect the rest of the console session after this point?

Reply:
resolving since this was replaced with `(New-Object System.Net.WebClient).DownloadFile($url, $outFile)`